# 2010 Minnesota Congressional Districts

Jfer09 opened the pull request:
## Redistricting requirements
In Minnesota, districts must:

1. be contiguous
2, have equal populations
3. comply with VRA section 2
4. be geographically compact
5. preserve political subdivisions and communities of interest as possible 
6. avoid pairing incumbents but also cannot give unfair advantage to incumbents (least important criteria)

https://www.mncourts.gov/mncourtsgov/media/CIOMediaLibrary/2011Redistricting/A110152Order11-4-11.pdf

### Interpretation of requirements
We do not adhere to all criteria in the guidelines. We include the following constraints:

1. We enforce a maximum population deviation of 0.5%. 
2. We use a pseudo-county constraint to help preserve county and municipality boundaries.

## Data Sources
Data for Minnesota comes from the ALARM Project's [2010 Redistricting Data Files](https://alarm-redist.github.io/posts/2021-08-10-census-2020/).

## Pre-processing Notes
No manual pre-processing decisions were necessary.

## Simulation Notes
We sample 5,000 districting plans  for Minnesota across two independent runs of the SMC algorithm.  We use a pseudo-county constraint to limit the county and municipality splits. Municipality lines used are in , which are all counties with populations larger than 60% the target population for district.

## Validation

![validation_20220923_0027](https://user-images.githubusercontent.com/66656384/191894568-e1f72f5e-6820-42ea-a93e-43458d115cd2.png)

```
SMC: 5,000 sampled plans of 8 districts on 4,139 units
`adapt_k_thresh`=0.985 • `seq_alpha`=0.5
`est_label_mult`=1 • `pop_temper`=0

Plan diversity 80% range: 0.75 to 0.96

R-hat values for summary statistics:
   pop_overlap      total_vap       plan_dev      comp_edge 
     1.0005790      1.0056396      1.0056415      1.0049476 
   comp_polsby      pop_white      pop_black       pop_hisp 
     1.0046960      1.0108363      0.9998412      1.0063150 
      pop_aian      pop_asian       pop_nhpi      pop_other 
     1.0104688      1.0001160      1.0035693      1.0027999 
       pop_two      vap_white      vap_black       vap_hisp 
     1.0010168      0.9999362      1.0004351      1.0057413 
      vap_aian      vap_asian       vap_nhpi      vap_other 
     1.0150546      1.0003509      1.0088918      1.0038647 
       vap_two pre_16_rep_tru pre_16_dem_cli pre_20_dem_bid 
     1.0030227      1.0056811      1.0035710      1.0015246 
pre_20_rep_tru uss_18_rep_new uss_18_dem_klo uss_20_dem_smi 
     1.0078803      1.0127675      1.0037961      1.0019473 
uss_20_rep_lew gov_18_rep_joh gov_18_dem_wal atg_18_rep_war 
     1.0245385      1.0107032      1.0064958      1.0110698 
atg_18_dem_ell sos_18_rep_how sos_18_dem_sim         adv_16 
     1.0057198      1.0118992      1.0041557      1.0035710 
        adv_18         adv_20         arv_16         arv_18 
     1.0051309      1.0012599      1.0056811      1.0117627 
        arv_20  county_splits    muni_splits            ndv 
     1.0109896      1.0090998      1.0011957      1.0032745 
           nrv        ndshare          e_dvs         pr_dem 
     1.0271921      1.0006879      1.0005442      1.0011941 
         e_dem          pbias           egap 
     0.9999207      1.0017316      1.0002931 

Sampling diagnostics for SMC run 1 of 2 (2,500 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique
Split 1     2,458 (98.3%)      9.8%        0.26 1,585 (100%)
Split 2     2,412 (96.5%)     13.8%        0.35 1,567 ( 99%)
Split 3     2,367 (94.7%)     17.6%        0.45 1,574 (100%)
Split 4     2,326 (93.1%)     19.4%        0.52 1,551 ( 98%)
Split 5     2,267 (90.7%)     17.1%        0.58 1,545 ( 98%)
Split 6     2,242 (89.7%)      7.8%        0.59 1,455 ( 92%)
Split 7     2,280 (91.2%)      4.4%        0.55 1,326 ( 84%)
Resample    1,647 (65.9%)       NA%        0.56 1,920 (121%)
         Est. k 
Split 1      13 
Split 2       8 
Split 3       5 
Split 4       4 
Split 5       4 
Split 6       7 
Split 7       4 
Resample     NA 

Sampling diagnostics for SMC run 2 of 2 (2,500 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique
Split 1     2,459 (98.4%)     11.6%        0.26 1,592 (101%)
Split 2     2,417 (96.7%)     15.8%        0.35 1,560 ( 99%)
Split 3     2,393 (95.7%)     18.5%        0.42 1,571 ( 99%)
Split 4     2,305 (92.2%)     20.0%        0.50 1,533 ( 97%)
Split 5     2,290 (91.6%)     16.4%        0.55 1,535 ( 97%)
Split 6     2,300 (92.0%)     17.4%        0.54 1,439 ( 91%)
Split 7     2,285 (91.4%)      4.5%        0.54 1,364 ( 86%)
Resample    1,583 (63.3%)       NA%        0.55 1,907 (121%)
         Est. k 
Split 1      11 
Split 2       7 
Split 3       5 
Split 4       4 
Split 5       4 
Split 6       3 
Split 7       4 
Resample     NA 
```

## Checklist

- [x] I have followed the [instructions](https://github.com/alarm-redist/fifty-states/blob/main/CONTRIBUTING.md)
- [x] I have updated the [tracker](https://docs.google.com/spreadsheets/d/1k_tYLoE49W_DCK1tcWbouoYZFI9WD76oayEt5TOmJg4/edit#gid=453387933)
- [x] All `TODO` lines from the template code have been removed
- [x] I have merged in the master branch and then recalculated summary statistics
- [x] I have run `enforce_style()` to format my code
- [x] The documentation copied above is up-to-date 
- [x] There are no data files in this pull request
- [x] None of the file output paths (for the `redist_map` and `redist_plans` objects, and summary statistics) have been edited

@CoryMcCartan
@christopherkenny

CoryMcCartan commented on analyses/MN_cd_2010/doc_MN_cd_2010.md:
```diff
@@ -0,0 +1,28 @@
+In Minnesota, districts must:
+
+1. be contiguous
+2, have equal populations
```
`,` -> `.`

Jfer09 replied:
fixed!

CoryMcCartan commented on analyses/MN_cd_2010/doc_MN_cd_2010.md:
```diff
@@ -0,0 +1,28 @@
+No manual pre-processing decisions were necessary.
+
+## Simulation Notes
+We sample 5,000 districting plans  for Minnesota across two independent runs of the SMC algorithm.  We use a pseudo-county constraint to limit the county and municipality splits. Municipality lines used are in , which are all counties with populations larger than 60% the target population for district.
```
maybe add a tiny bit more detail here (see some other state PRs for examples) to explain the pseudocounty constraint.

As far as getting the counties themselves, the numbers you get out of the R code when you run it like `037` are county FIPS codes.  IF you look up "MN county 037" you'll see the name that corresponds to the code.  These are also stored in `tigris::fips_codes` if you want a programmatic solution

CoryMcCartan commented on analyses/SC_cd_2010/03_sim_SC_cd_2010.R:
```diff
@@ -92,4 +92,14 @@ if (interactive()) {
         group_by(draw) %>%
         summarize(n_black_perf = sum(vap_black/total_vap > 0.3 & ndshare > 0.5)) %>%
         count(n_black_perf)
+
+    redist.plot.hist(plans = plans_5k, qty = e_dem) +
+        # scale_x_continuous(name = 'Expected Number of Democratic Districts') +
+        theme_bw()
+
+    redist.plot.plans(plans, draw = 100, shp = map, qty = ndv/(ndv + nrv), ) +
+        # scale_fill_party_c() +
+        # theme_map() +
+        theme(legend.position = "right") +
+        labs(title = "")
```
Hm I don't think this should be in this branch. Can you revert this change?

CoryMcCartan commented on analyses/MN_cd_2010/01_prep_MN_cd_2010.R:
```diff
@@ -0,0 +1,87 @@
+        invisible()
+
+    # simplifies geometry for faster processing, plotting, and smaller shapefiles
+    # TODO feel free to delete if this dependency isn't available
```
remove this TODO line

Jfer09 replied:
fixed!

CoryMcCartan commented on analyses/MN_cd_2010/02_setup_MN_cd_2010.R:
```diff
@@ -0,0 +1,22 @@
+attr(map, "analysis_name") <- "MN_2010"
+
+# Fix labeling
+map$state <- "MN"
```
can you move this to the prep file? no need to re-run any other code, just I think it belongs there

CoryMcCartan reviewed (changes requested):
Sims look great! Some small things to fix re: code and documentation but then we'll be good to merge. @christopherkenny let us know if there's anything else you see

Jfer09 commented:
## Redistricting requirements
In Minnesota, districts must:

1. be contiguous
2. have equal populations
3. comply with VRA section 2
4. be geographically compact
5. preserve political subdivisions and communities of interest as possible 
6. avoid pairing incumbents but also cannot give unfair advantage to incumbents (least important criteria)

https://www.mncourts.gov/mncourtsgov/media/CIOMediaLibrary/2011Redistricting/A110152Order11-4-11.pdf

### Interpretation of requirements
We do not adhere to all criteria in the guidelines. We include the following constraints:

1. We enforce a maximum population deviation of 0.5%. 
2. We use a pseudo-county constraint to help preserve county and municipality boundaries.

## Data Sources
Data for Minnesota comes from the ALARM Project's [2010 Redistricting Data Files](https://alarm-redist.github.io/posts/2021-08-10-census-2020/).

## Pre-processing Notes
No manual pre-processing decisions were necessary.

## Simulation Notes
We sample 5,000 districting plans for Minnesota across two independent runs of the SMC algorithm. To balance county and municipality splits, we create pseudocounties for use in the county constraint. These are counties, outside of Dakota County, Hennepin County, Ramsey, which are the counties with populations larger than 60% the target population for districts. Within Allegheny County, Montgomery County, and Philadelphia County, each municipality is its own pseudocounty as well.

## Validation

![validation_20220923_0027](https://user-images.githubusercontent.com/66656384/191894568-e1f72f5e-6820-42ea-a93e-43458d115cd2.png)

```
SMC: 5,000 sampled plans of 8 districts on 4,139 units
`adapt_k_thresh`=0.985 • `seq_alpha`=0.5
`est_label_mult`=1 • `pop_temper`=0

Plan diversity 80% range: 0.75 to 0.96

R-hat values for summary statistics:
   pop_overlap      total_vap       plan_dev      comp_edge 
     1.0005790      1.0056396      1.0056415      1.0049476 
   comp_polsby      pop_white      pop_black       pop_hisp 
     1.0046960      1.0108363      0.9998412      1.0063150 
      pop_aian      pop_asian       pop_nhpi      pop_other 
     1.0104688      1.0001160      1.0035693      1.0027999 
       pop_two      vap_white      vap_black       vap_hisp 
     1.0010168      0.9999362      1.0004351      1.0057413 
      vap_aian      vap_asian       vap_nhpi      vap_other 
     1.0150546      1.0003509      1.0088918      1.0038647 
       vap_two pre_16_rep_tru pre_16_dem_cli pre_20_dem_bid 
     1.0030227      1.0056811      1.0035710      1.0015246 
pre_20_rep_tru uss_18_rep_new uss_18_dem_klo uss_20_dem_smi 
     1.0078803      1.0127675      1.0037961      1.0019473 
uss_20_rep_lew gov_18_rep_joh gov_18_dem_wal atg_18_rep_war 
     1.0245385      1.0107032      1.0064958      1.0110698 
atg_18_dem_ell sos_18_rep_how sos_18_dem_sim         adv_16 
     1.0057198      1.0118992      1.0041557      1.0035710 
        adv_18         adv_20         arv_16         arv_18 
     1.0051309      1.0012599      1.0056811      1.0117627 
        arv_20  county_splits    muni_splits            ndv 
     1.0109896      1.0090998      1.0011957      1.0032745 
           nrv        ndshare          e_dvs         pr_dem 
     1.0271921      1.0006879      1.0005442      1.0011941 
         e_dem          pbias           egap 
     0.9999207      1.0017316      1.0002931 

Sampling diagnostics for SMC run 1 of 2 (2,500 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique
Split 1     2,458 (98.3%)      9.8%        0.26 1,585 (100%)
Split 2     2,412 (96.5%)     13.8%        0.35 1,567 ( 99%)
Split 3     2,367 (94.7%)     17.6%        0.45 1,574 (100%)
Split 4     2,326 (93.1%)     19.4%        0.52 1,551 ( 98%)
Split 5     2,267 (90.7%)     17.1%        0.58 1,545 ( 98%)
Split 6     2,242 (89.7%)      7.8%        0.59 1,455 ( 92%)
Split 7     2,280 (91.2%)      4.4%        0.55 1,326 ( 84%)
Resample    1,647 (65.9%)       NA%        0.56 1,920 (121%)
         Est. k 
Split 1      13 
Split 2       8 
Split 3       5 
Split 4       4 
Split 5       4 
Split 6       7 
Split 7       4 
Resample     NA 

Sampling diagnostics for SMC run 2 of 2 (2,500 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique
Split 1     2,459 (98.4%)     11.6%        0.26 1,592 (101%)
Split 2     2,417 (96.7%)     15.8%        0.35 1,560 ( 99%)
Split 3     2,393 (95.7%)     18.5%        0.42 1,571 ( 99%)
Split 4     2,305 (92.2%)     20.0%        0.50 1,533 ( 97%)
Split 5     2,290 (91.6%)     16.4%        0.55 1,535 ( 97%)
Split 6     2,300 (92.0%)     17.4%        0.54 1,439 ( 91%)
Split 7     2,285 (91.4%)      4.5%        0.54 1,364 ( 86%)
Resample    1,583 (63.3%)       NA%        0.55 1,907 (121%)
         Est. k 
Split 1      11 
Split 2       7 
Split 3       5 
Split 4       4 
Split 5       4 
Split 6       3 
Split 7       4 
Resample     NA 
```

## Checklist

- [x] I have followed the [instructions](https://github.com/alarm-redist/fifty-states/blob/main/CONTRIBUTING.md)
- [x] I have updated the [tracker](https://docs.google.com/spreadsheets/d/1k_tYLoE49W_DCK1tcWbouoYZFI9WD76oayEt5TOmJg4/edit#gid=453387933)
- [x] All `TODO` lines from the template code have been removed
- [x] I have merged in the master branch and then recalculated summary statistics
- [x] I have run `enforce_style()` to format my code
- [x] The documentation copied above is up-to-date 
- [x] There are no data files in this pull request
- [x] None of the file output paths (for the `redist_map` and `redist_plans` objects, and summary statistics) have been edited

@CoryMcCartan
@christopherkenny

Jfer09 commented on analyses/MN_cd_2010/doc_MN_cd_2010.md:
```diff
@@ -0,0 +1,28 @@
+No manual pre-processing decisions were necessary.
+
+## Simulation Notes
+We sample 5,000 districting plans for Minnesota across two independent runs of the SMC algorithm. To balance county and municipality splits, we create pseudocounties for use in the county constraint. These are counties, outside of Dakota County, Hennepin County, Ramsey, which are the counties with populations larger than 60% the target population for districts. Within Allegheny County, Montgomery ...
```
Just changed and pushed! Thanks so much!!
